Charts panel open/close toggle

uniphil opened the pull request.

uniphil committed:
Add open/closed state for charts

uniphil committed:
Animate charts open/close

Requires adding componentShouldUpdate to WaterpointMarker so they don't all re-render

esoliani commented on app/scripts/components/dashboard/charts-container.jsx:
```diff
@@ -7,16 +7,30 @@ require('stylesheets/dashboard/charts-container');
+                role="button">
+              <T k="charts.toggle.activate" />
+            </span>
+            <ViewMode />
```
ViewMode should be ouside of the charts-containter-header, shouldn't it?
Let's leave it and change it on the future

uniphil replied:
yeah hard to name things... `header` was meant as the nav-bar at the top of charts-container, so maybe it should be `charts-container-nav`?

esoliani commented on app/scripts/stores/layout.es6:
```diff
@@ -0,0 +1,30 @@
+  },
+  update(what, to) {
+    this.setData({
+      ...this.get(),
```
I'm not used to this notation, and it means different on es6 and jsx

uniphil replied:
It's sort of different, and sort of the same. Here is what the JSX compiles down to with the object spread operator:

``` js
const otherProps = {a: 1, b: 2};
<Component prop="val" prop2="val2" {...otherProps} />
```

compiles to

``` js
React.createElement(Component, {
  prop: 'val',
  prop2: 'val2',
  ...otherProps
}, []);
```

so it's.... kind of... the same... ish

The nice thing about it is that it's a short way to create a new object with some things changed from the old one without mutating it. I'm going to pull in immutablejs though, which will enforce that and give us a more normal javascript syntax (and performance improvements and awesome stuff).

uniphil replied:
I will switch to immutablejs in a separate PR

esoliani commented on app/scripts/components/dashboard/waterpoints.jsx:
```diff
@@ -40,13 +43,12 @@ const WaterPoints = React.createClass({
+        <ChartsContainer
+            onToggle={toggleCharts}
+            state={this.state.layout.charts}>
           <MetricStatus metric="54.65" title="chart.title.functional" total="123456"/>
```
I think these should be always visible.
In here we should add new component with chats

uniphil replied:
yep you're right. I'll fix it.

uniphil committed:
Rename charts-container-header => -nav, to be clearer

Still not a great name. Maybe the whole thing should be renamed to... `Bottom`? I don't know the word for `sidebar`
that is at the bottom.

uniphil committed:
Move metricStatus out of charts body so it is visible when they are closed.

uniphil commented:
thanks!!

On Thu, Oct 1, 2015 at 8:47 AM, Ernesto Soliani notifications@github.com
wrote:

> Merged #12 https://github.com/WorldBank-Transport/waterdash/pull/12.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/WorldBank-Transport/waterdash/pull/12#event-424178262
> .
